PowerEditor/src/localization.cpp

```diff
@@ -388,25 +388,31 @@ static const int tabContextMenuItemPos[] =
     5,   //  2: Save
     6,   //  3: Save As
    10,   //  4: Print
-   25,   //  5: Move to Other View
-   26,   //  6: Clone to Other View
-   21,   //  7: Full File Path to Clipboard
-   22,   //  8: Filename to Clipboard
-   23,   //  9: Current Dir. Path to Clipboard
-    7,   // 10: Rename
-    8,   // 11: Move to Recycle Bin
-   18,   // 12: Read-Only
-   19,   // 13: Clear Read-Only Flag
-   27,   // 14: Move to New Instance
-   28,   // 15: Open to New Instance
-    9,   // 16: Reload
-    2,   // 17: Close ALL to the Left
-    3,   // 18: Close ALL to the Right
-   12,   // 19: Open Containing Folder in Explorer
-   13,   // 20: Open Containing Folder in cmd
-   16,   // 21: Open in Default Viewer
-    4,   // 22: Close ALL Unchanged
-   14,   // 23: Open Containing Folder as Workspace
+   30,   //  5: Apply Color 1
+   31,   //  6: Apply Color 2
+   32,   //  7: Apply Color 3
+   33,   //  8: Apply Color 4
+   34,   //  9: Apply Color 5
+   35,   // 10: Remove Color
+   25,   // 11: Move to Other View
+   26,   // 12: Clone to Other View
+   21,   // 13: Full File Path to Clipboard
+   22,   // 14: Filename to Clipboard
+   23,   // 15: Current Dir. Path to Clipboard
+    7,   // 16: Rename
+    8,   // 17: Move to Recycle Bin
+   18,   // 18: Read-Only
+   19,   // 19: Clear Read-Only Flag
+   27,   // 20: Move to New Instance
+   28,   // 21: Open to New Instance
+    9,   // 22: Reload
+    2,   // 23: Close ALL to the Left
+    3,   // 24: Close ALL to the Right
+   12,   // 25: Open Containing Folder in Explorer
+   13,   // 26: Open Containing Folder in cmd
+   16,   // 27: Open in Default Viewer
+    4,   // 28: Close ALL Unchanged
+   14,   // 29: Open Containing Folder as Workspace

```

Comment:
The node `<TabBar>` in english.xml should be replaced by the following xml code:

```xml
            <TabBar>
                    <Item CMID="0" name="Close"/>
                    <Item CMID="1" name="Close All BUT This"/>
                    <Item CMID="2" name="Save"/>
                    <Item CMID="3" name="Save As..."/>
                    <Item CMID="4" name="Print..."/>
                    <Item CMID="5" name="Move to Other View"/>
                    <Item CMID="6" name="Clone to Other View"/>
                    <Item CMID="7" name="Full File Path to Clipboard"/>
                    <Item CMID="8" name="Filename to Clipboard"/>
                    <Item CMID="9" name="Current Dir. Path to Clipboard"/>
                    <Item CMID="10" name="Rename..."/>
                    <Item CMID="11" name="Move to Recycle Bin"/>
                    <Item CMID="12" name="Read-Only"/>
                    <Item CMID="13" name="Clear Read-Only Flag"/>
                    <Item CMID="14" name="Move to New Instance"/>
                    <Item CMID="15" name="Open in New Instance"/>
                    <Item CMID="16" name="Reload"/>
                    <Item CMID="17" name="Close All to the Left"/>
                    <Item CMID="18" name="Close All to the Right"/>
                    <Item CMID="19" name="Open Containing Folder in Explorer"/>
                    <Item CMID="20" name="Open Containing Folder in cmd"/>
                    <Item CMID="21" name="Open in Default Viewer"/>
                    <Item CMID="22" name="Close All Unchanged"/>
                    <Item CMID="23" name="Open Containing Folder as Workspace"/>
                    <Item CMID="24" name="Apply Color 1"/>
                    <Item CMID="25" name="Apply Color 2"/>
                    <Item CMID="26" name="Apply Color 3"/>
                    <Item CMID="27" name="Apply Color 4"/>
                    <Item CMID="28" name="Apply Color 5"/>
                    <Item CMID="29" name="Remove Color"/>
            </TabBar>
```

Then you replace whole `static const int tabContextMenuItemPos[] ={...]` by:

```cpp
static const int tabContextMenuItemPos[] =
{
	//  +-------------- The item positon in tab context menu
	//  |
	//  |        +------ Number in english.xml : <TabBar> CMDID
	//  |        |
        0,   //  0: Close
        1,   //  1: Close ALL BUT This
        5,   //  2: Save
        6,   //  3: Save As
       10,   //  4: Print
       32,   //  5: Move to Other View
       33,   //  6: Clone to Other View
       28,   //  7: Full File Path to Clipboard
       29,   //  8: Filename to Clipboard
       30,   //  9: Current Dir. Path to Clipboard
        7,   // 10: Rename
        8,   // 11: Move to Recycle Bin
       25,   // 12: Read-Only
       24,   // 13: Clear Read-Only Flag
       34,   // 14: Move to New Instance
       35,   // 15: Open to New Instance
        9,   // 16: Reload
        2,   // 17: Close ALL to the Left
        3,   // 18: Close ALL to the Right
       19,   // 19: Open Containing Folder in Explorer
       20,   // 20: Open Containing Folder in cmd
       23,   // 21: Open in Default Viewer
        4,   // 22: Close ALL Unchanged
       21,   // 23: Open Containing Folder as Workspace
       12,   // 24: Apply Color 
       13,   // 25: Apply Color 
       14,   // 26: Apply Color
       15,   // 27: Apply Color
       16,   // 28: Apply Color
       17,   // 29: Remove Color
       -1    //-------End
};
```

The comment of `tabContextMenuItemPos` wasn't clear. I have modified comment (above).
Each time if we add a new command, we have to "push back" its localization entry into `<TabBar>` in english.xml, so it's the index order of new command for localization. The index order of old commands keep the same.

Then we have to also "push back" new command's position into `tabContextMenuItemPos` and change the position of other commands, according where you insert the new command.

 Both array keep synchronized so Notepad++ know replace the label of menu item by localization string at the indicated position.